examples/petsc/dmswarm.c

```diff
@@ -0,0 +1,410 @@
+      point_cell_numbers[cell_points[p]] = cell;
+      for (PetscInt d = 0; d < dim; d++) coords_cell_points_true[p * dim + d] = coords_points_true[cell_points[p] * dim + d];
+    }
+    PetscCall(DMPlexCoordinatesToReference(dm_mesh, cell, num_points_in_cell, coords_cell_points_true, coords_cell_points_ref));
+    for (PetscInt p = 0; p < num_points_in_cell; p++) {
+      for (PetscInt d = 0; d < dim; d++) coords_points_ref[cell_points[p] * dim + d] = coords_cell_points_ref[p * dim + d];
+    }
+
+    // -- Cleanup
+    PetscCall(DMRestoreWorkArray(dm_mesh, num_points_in_cell * dim, MPIU_REAL, &coords_cell_points_true));
+    PetscCall(DMRestoreWorkArray(dm_mesh, num_points_in_cell * dim, MPIU_REAL, &coords_cell_points_ref));
+    PetscCall(PetscFree(cell_points));
+  }
+
+  // Cleanup
+  PetscCall(DMSwarmRestoreField(dm_swarm, "DMSwarmPIC_coor", NULL, NULL, (void **)&coords_points_ref));
+  PetscCall(VecRestoreArrayRead(*ref_coords, &coords_points_true));
+  PetscCall(DMSwarmSortRestoreAccess(dm_swarm));
+
+  // Create index set
+  PetscCall(ISCreateGeneral(PETSC_COMM_SELF, num_points_local, point_cell_numbers, PETSC_OWN_POINTER, is_points));
+
+  PetscFunctionReturn(PETSC_SUCCESS);
+}
+
+int main(int argc, char **argv) {
+  MPI_Comm            comm;
+  DM                  dm_mesh, dm_swarm;
+  Vec                 U_mesh;
+  PetscInt            dim = 3, num_comp = 1, num_points = 10, geometry_order = 1, mesh_order = 2, q_order = 3;
+  Ceed                ceed;
+  CeedBasis           basis_mesh_u;
+  CeedElemRestriction restriction_mesh_u;
+
+  PetscCall(PetscInitialize(&argc, &argv, NULL, help));
+  comm = PETSC_COMM_WORLD;
+  CeedInit("/cpu/self", &ceed);
+
+  // Create background mesh
+  PetscCall(DMCreate(comm, &dm_mesh));
+  PetscCall(DMSetType(dm_mesh, DMPLEX));
+  PetscCall(DMSetFromOptions(dm_mesh));
+
+  // -- Mesh FE space
+  {
+    PetscFE fe;
+
+    PetscCall(DMGetDimension(dm_mesh, &dim));
+    PetscCall(PetscFECreateLagrange(comm, dim, num_comp, PETSC_FALSE, mesh_order, q_order, &fe));
+    PetscCall(DMAddField(dm_mesh, NULL, (PetscObject)fe));
+    PetscCall(PetscFEDestroy(&fe));
+  }
+  PetscCall(DMCreateDS(dm_mesh));
+
+  // -- Coordinate FE space
+  {
+    PetscFE fe_coord;
+
+    PetscCall(PetscFECreateLagrange(comm, dim, num_comp, PETSC_FALSE, geometry_order, q_order, &fe_coord));
+    PetscCall(DMProjectCoordinates(dm_mesh, fe_coord));
+    PetscCall(DMProjectCoordinates(dm_mesh, fe_coord));
+    PetscCall(PetscFEDestroy(&fe_coord));
+  }
+
+  // -- Final background mesh
+  PetscCall(PetscObjectSetName((PetscObject)dm_mesh, "Background Mesh"));
+  PetscCall(DMViewFromOptions(dm_mesh, NULL, "-dm_mesh_view"));
+
+  // -- libCEED objects from background mesh
+  PetscCall(CreateBasisFromPlex(ceed, dm_mesh, NULL, 0, 0, 0, /* Don't want BP data here */ bp_options[CEED_BP1], &basis_mesh_u));
+  PetscCall(CreateRestrictionFromPlex(ceed, dm_mesh, 0, NULL, 0, &restriction_mesh_u));
```

Comment:
These functions aren't linking correctly. Also, I don't really want to have to include the BP data just for the purpose of including it here. `CreateBasisFromPlex` should be updated to accept a `NULL` BP data and use standard quadrature in that case.